proctl/breakpoints_linux_amd64.go

```diff
@@ -133,24 +143,49 @@ func (dbp *DebuggedProcess) newBreakpoint(fn, f string, l int, addr uint64, data
 // that we want to break at. There are only 4 debug registers
 // DR0-DR3. Debug register 7 is the control register.
 func setHardwareBreakpoint(reg, tid int, addr uint64) error {
-	if reg < 0 || reg > 7 {
+	if reg < 0 || reg > 3 {
```

Comment:
Yeah, this is good. Don't know what I was thinking here... out of the 7 debug regs only the 4 are used for actually storing the address.